### ISSUE TITLE
When reporting build errors in multiple modules, sort modules either by last modification or dependency graph.

### ISSUE THREAD

**Issue by AndreasPK** [haskell/haskell-language-server#3278]
## Is your enhancement request related to a problem? Please describe.

Currently the ordering of *Modules* for error reporting can require repeated scrolling.

The situation is as follows:

* A project structure where `Main` depends on `Parser`.
* I change a type in the parser code.
* The change causes a new error in Main while there are still active type errors in the parser.
* HLS in VScode will show me first the type errors in Main (which are transient, and currently useless) before listing the errors in Parser.hs further down.

## Describe the solution you'd like

Ideally we would have a magic oracle that tells us which module a user is currently working on, and errors in that module would be shown first. Usually VSCode is pretty good at this but I think in the combination that a change causes a *new* error in Main but adds no error in Parser it starts showing the error list for Main.

## Describe alternatives you've considered

I usually do refactorings from the root of the dependency graph downwards so this behaviour is'nt helpful to me. But perhaps others find it helpful? I'm also not sure if HLS can even control the ordering.

**Comment by michaelpj** [comment]
I don't think this is anything we can control? We just report diagnostics to the client and it's up to the client how to show them to the user. Maybe VSCode puts the most "recent" diagnostics at the top, but I doubt this is specified behaviour so I don't think we can rely on it...
